static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -238,11 +243,55 @@ function WidgetViewerModal(props: Props) {
     displayType: DisplayType.TABLE,
   };
   const {aggregates, columns} = tableWidget.queries[0];
+  const {orderby} = widget.queries[0];
+  const order = orderby.startsWith('-');
+  const rawOrderby = trimStart(orderby, '-');
 
   const fields = defined(tableWidget.queries[0].fields)
     ? tableWidget.queries[0].fields
     : [...columns, ...aggregates];
 
+  // Some Discover Widgets (Line, Area, Bar) allow the user to specify an orderby
+  // that is not explicitly selected as an aggregate or column. We need to explictly
+  // include the orderby in the table widget aggregates and columns otherwise
+  // eventsv2 will complain about sorting on an unselected field.
+  if (
+    widget.widgetType === WidgetType.DISCOVER &&
+    orderby &&
+    !fields.includes(rawOrderby)
+  ) {
+    fields.push(rawOrderby);
+    [tableWidget, primaryWidget].forEach(aggregatesAndColumns => {
```

Comment:
Is there a particular reason we add it to the `primaryWidget` here? I think this is the reason the widget state gets updated and I don't think it's necessary. The orderby gets handled in `widgetQueries.tsx` if it's not in the fields. 

I like keeping this logic here though

Reply:
Yep, makes total sense now that you mention the custom equations. Thanks for clarifying that!